### PR TITLE
Prepare

### PR DESCRIPTION
**Pull request by GennadyBerg**
to run  "npm run dev"

### CODE REVIEW

File: businesslogic/usersAuth.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+const users = [];
+
+function signupUser(req, res) {
+  {
+    
+    const { username, email, password } = req.body;
+
+    
+    if (users.some(user => user.username === username)) {
+      return res.status(400).json({ error: 'Username is already taken' });
```
Comment (iKornev) [inline businesslogic/usersAuth.js]: here is better to throw error or call next() with error

File: businesslogic/usersAuth.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    return res.status(400).json({ error: error.details[0].message });
+  }*/
+
+  const { username, password } = req.body;
+
+ 
+  const user = users.find(user => user.username === username);
+
+  if (!user || !bcrypt.compareSync(password, user.password)) {
+    return res.status(401).json({ error: 'Invalid username or password' });
```
Comment (iKornev) [inline businesslogic/usersAuth.js]: here is better to throw error or call next() with error

File: businesslogic/usersAuth.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    users.push({ username, email, password: hashedPassword });
+
+    return res.status(201).json({ message: 'Registration successful' });
+  }
+}
+
+
+function signinUser(req, res) {
+ 
+  if (error) {
```
Comment (iKornev) [inline businesslogic/usersAuth.js]: what is it?
how error can appear here ?

File: businesslogic/usersAuth.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    return res.status(400).json({ error: error.details[0].message });
+  }*/
```
Comment (iKornev) [inline businesslogic/usersAuth.js]: what is it ? 
spoiler: this is an error that occurs during server startup
```
file:///Users/ivankornev/Desktop/Students%20HW/Gennadiy%20Shterenberg/HW7/businesslogic/usersAuth.js:30
  }*/
   ^

SyntaxError: Unexpected token '*'
    at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:167:18)
    at callTranslator (node:internal/modules/esm/loader:285:14)
    at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:291:30)

Node.js v20.11.0
[nodemon] app crashed - waiting for file changes before starting...
```

File: validationSchemas/userValidationSchema.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    });
+    
+///////
+
+const validateCreationSchema = Joi.object({
+      requests: Joi.array().items(
+        Joi.object({
+          line: Joi.number().required(),
+          valve: Joi.number(),
+          valves: Joi.array().when('valve', { is: Joi.exist(), then: Joi.required() }),
```
Comment (iKornev) [inline validationSchemas/userValidationSchema.js]: what will happen if I remove `valves` and will use only `valve` 
hint:  `Joi.alternatives()`

File: businesslogic/usersAuth.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  const user = users.find(user => user.username === username);
+
+  if (!user || !bcrypt.compareSync(password, user.password)) {
+    return res.status(401).json({ error: 'Invalid username or password' });
+  }
+
+  return res.status(200).json({ message: 'Authentication successful' });
+}
+
+function validateUser(req, res) {
```
Comment (iKornev) [inline businesslogic/usersAuth.js:39]: what is this ?
 for what do you use this function ?

File: businesslogic/usersAuth.js (hunk 1 above)
Comment (GennadyBerg) [reply businesslogic/usersAuth.js]: You are right! next() is the better

File: businesslogic/usersAuth.js (hunk 6 above)
Comment (GennadyBerg) [reply businesslogic/usersAuth.js:39]: for examle object validation (userRouter : line 10)
userRouter.post('/validateObject', validator(validateCreationSchema), validateUser);

File: validationSchemas/userValidationSchema.js (hunk 5 above)
Comment (GennadyBerg) [reply validationSchemas/userValidationSchema.js]: than I need use  Joi.alternatives()  ?
no problem

File: businesslogic/usersAuth.js (hunk 4 above)
Comment (GennadyBerg) [reply businesslogic/usersAuth.js]: it's  my technical mistake when I removed some lines of modules))))
sorry

**Pull request merged**